### DISCUSSION TITLE
Remove overlap produces leftover paths

### DISCUSSION THREAD

**Discussion by gorjious** [coldtype/coldtype#145] [Q&A] [answered]
So I'm trying to use Coldtype to recreate the ripple idea of [Typographics 2018](https://2018.typographics.com/). 

I got it mostly working but whenever I increase size or number of ripples then all these small paths that seem to be leftover paths from the remove overlap get exaggerated and create noise.

Got bit of a roundabout way of doing which I'm sure can be improved. I thought that by maybe making the first ripple with .union() and the using that as a basis for creating the rest of the ripples might make it cleaner. Then I noticed that .outline() was creating inner and outer shapes...for some reason when setting drawOuter=False felt counterintuitive but seems to be working in the right direction. I was trying all sorts of path operations and .removeOverlap but to no avail. 

Maybe you can take a peek and offer some insights and improvements?

Here is the script: [hypnotic_patches_coldtype_1.py.zip](https://github.com/coldtype/coldtype/files/12839786/hypnotic_patches_coldtype_1.py.zip)

Drawing that first ripple and then using .removeOvelap or .union() produces these leftover segments. How can I get rid of them? Maybe some sort of check for paths within outer path and delete?

<img width="849" alt="Screenshot 2023-10-07 at 11 08 44 PM" src="https://github.com/coldtype/coldtype/assets/23149420/f05f7306-eba0-456d-8f05-b14edcfa1f82">

<img width="854" alt="Screenshot 2023-10-07 at 11 09 33 PM" src="https://github.com/coldtype/coldtype/assets/23149420/9a339668-9dbb-4105-a6f3-f2012f52909d">

Then the issue becomes apparent when more ripples get added.

<img width="882" alt="Screenshot 2023-10-07 at 10 25 51 PM" src="https://github.com/coldtype/coldtype/assets/23149420/1c61b420-1806-4199-a49e-ed2e8447c9ae">

<img width="895" alt="Screenshot 2023-10-07 at 10 23 47 PM" src="https://github.com/coldtype/coldtype/assets/23149420/02747a84-4556-4374-bf2f-3780465ee114">

<img width="943" alt="Screenshot 2023-10-07 at 10 23 59 PM" src="https://github.com/coldtype/coldtype/assets/23149420/c7637daa-c02e-4941-8df1-a802a7b6e2b6">

**Comment by stenson** [comment, marked as answer]
Cool! Yeah I remember the 2018 Typographics graphics — I think I remember reading somewhere it's actually an interpolation-based scheme, rather than a "growing" outline, i.e. the rectangle at the edge is an interpolatable rectangle of the text in the middle, altho now that I said that, I'm not sure if that looks right when I'm looking at the actual graphics.

Anyway, here's a version of the same idea as in your code — I'm not getting the overlaps you're seeing although I'm not sure what the difference in the approach is:

```python
from coldtype import *

@renderable((1080, 540), bg=1)
def hypnotic(r):
    text = "Outline\ngraphics"
    
    basis = StSt(text, "Megazoid-Regular", 100).align(r)
    slug = StSt(text, "Megazoid-Fill", 100).align(r).pen().removeOverlap()
    
    out = P()
    for x in range(1, 60):
        out.append(slug.copy().outline(4*x, drawOuter=False).removeOverlap())
    
    frame = P(r.inset(100))

    return (P(
        basis.pen().outline(1),
        out.pen().outline(1).intersection(frame),
        frame.outline(1))
        .f(0))
```

![hypnotic](https://github.com/coldtype/coldtype/assets/89770/33df754f-f448-4c5f-9752-18c0aeb8227e)

**Comment by gorjious** [comment]
Yeah I just came across a bit about the interpolation on Andy Clymer's [site](http://www.andyclymer.com/recently/) . Has such strong graphic impact no matter how you slice it up!

Not quite sure of the difference but I think had something to do with how and the order that union() and removeOverlap() was being applied. 

Cleaned it up some based on your perfectly concise example and now I'm not seeing the leftoverlaps.
Thank you!
